### ISSUE TITLE
Office.context.mailbox.item.addFileAttachmentAsync() fails in Outlook Web on-prem (Exchange Server 2019 v15.2, build 1544.4)

### ISSUE THREAD

**Issue by GruberMarkus** [OfficeDev/office-js#4926]
# Provide required information needed to triage your issue

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web (on premises)
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: Outlok Web on Exchange Server 2019 (version 15.2, build 1544.4)
* Operating System: Multiple (Windows 11 latest, Ubuntu 22.04, macOs Sonoma latest)
* Browser (if using Office on the web): Edge (latest version), Firefox (latest version)
 
## Expected behavior
Office.context.mailbox.item.addFileAttachmentAsync() should add an attachment.

## Current behavior
When calling Office.context.mailbox.item.addFileAttachmentAsync(), the folloing error is returned: {"name":"Host Error","message":"The operation is not supported.","code":5000}

## Steps to reproduce

1. Create a taskpane add-in for Outlook
2. Add a file using `Office.context.mailbox.item.addFileAttachmentAsync(<public URL to image file>, "my file name.jpg", { isInline: false})`
3. Deploy add-in
4. Run in Outlook Web on premises

## Link to live example(s)
Not available

# Provide additional details
The very same code works fine in Outlook Web running on Exchange Online, as well as in Classic Outlook on Windows. Other platforms have not been tested.

Please let me know how to avoid this error. addFileAttachementFromBase64Asynch() is not an option, as it is not support in Outlook Web on premises.

Edit: Replace Office.context.mailbox.item.**addFileAttachmentFromBase64Async()** with Office.context.mailbox.item.**addFileAttachmentAsync()** in sample code and error description.

**Comment by Oleg-O** [comment]
This API was added in requirement set 1.8; but per the support table on [this page](https://learn.microsoft.com/en-us/javascript/api/requirement-sets/outlook/outlook-api-requirement-sets?view=common-js-preview&tabs=jsonmanifest), Outlook Web on-premises supports up to and including 1.6 only, so this is expected.

**Comment by GruberMarkus** [comment]
According to the documentation of the method at https://learn.microsoft.com/en-us/javascript/api/outlook/office.messagecompose?view=outlook-js-preview&preserve-view=true#outlook-office-messagecompose-addfileattachmentasync-member(1), the required API is 1.1.

From what I see in the API 1.8 release notes, addFileAttachmentFromBase64Async has been introduced with 1.8, but not addFileAttachmentAsync (without Base64).

I mixed up these two myself in the bug description. Sorry for that. 

The problem is with Office.context.mailbox.item.addFileAttachmentAsync().

Could you please have another look at this topic?

Edit: I replaced Office.context.mailbox.item.**addFileAttachmentFromBase64Async()** with Office.context.mailbox.item.**addFileAttachmentAsync()** in sample code and error description of the opening comment.

**Comment by Swathy-Mothilal** [comment]
Hi @GruberMarkus, Could you please let me know if this was functioning as expected previously and since when are you experiencing the issue?

**Comment by GruberMarkus** [comment]
No, I can't. I am only using Office.context.mailbox.item.addFileAttachmentAsync() since a few days.

**Comment by Swathy-Mothilal** [comment]
@GruberMarkus Thanks for the confirmation. We are able to repro the bug, and have put it in our backlog. Unfortunately, we don't have any timelines to share.

Internal tracking id: 4944585

**Comment by whoanuragverma** [comment]
The minimum requirement set was updated to 1.8 for `addFileAttachmentAsync()`, we are in the process of updating our public documentation. Exchange on-prem 2019 supports only upto 1.5, so this is expected.

Internal Tracking ID: 314341

**Comment by GruberMarkus** [comment]
@whoanuragverma: So there is no way to add an attachment, such as an inline image, to an email in Outlook Web on-prem? How do tools like Exclaimer and CodeTwo then do exactly this?

https://learn.microsoft.com/en-us/javascript/api/requirement-sets/outlook/preview-requirement-set/office.context.mailbox.item?view=common-js-preview#methods and https://learn.microsoft.com/en-us/javascript/api/outlook/office.messagecompose?view=outlook-js-preview&preserve-view=true#outlook-office-messagecompose-addfileattachmentasync-member(1) still list 1.1 as the minimum requirement set for addFileAttachmentAsync.

**Comment by neprasad-microsoft** [comment]
Hi @GruberMarkus 
This change was done to overcome a security vulnerability in the original API. You can read more about it here- https://www.zerodayinitiative.com/advisories/ZDI-23-1581/ . 
Thank you for bringing this to our attention. We are currently updating our public docs to represent the right behaviour. Alternatively, you can use Graph APIs to attach files-https://learn.microsoft.com/en-us/graph/api/message-post-attachments?view=graph-rest-1.0&tabs=http.

**Comment by GruberMarkus** [comment]
@neprasad-microsoft, Graph APIs do not work on-prem.

Do I assume correctly that since the end of 2023, there is no way for an Outlook add-in to add images in Outlook Web on-prem?

**Comment by neprasad-microsoft** [comment]
Hi @GruberMarkus 
You can try the below code to add an attachment using HTML embedding in the body-

`Office.context.mailbox.item.body.setAsync(
			"<p>Here's a cute bird!</p><img src='[http://i.imgur.com/WJXklif.png'>"](http://i.imgur.com/WJXklif.png%27%3E%22),
			{
				"coercionType": "html"
			},
			function (asyncResult) {
			}
		);`

**Comment by GruberMarkus** [comment]
@neprasad-microsoft: That's linking a file, not embedding, and it certainly is not adding a file as inline attachment.

Instead of repairing addFileAttachmentAsync, it has basically been removed from on-prem without announcement, documentation and a real alternative.

**Comment by DivyaPatidar** [comment]
@GruberMarkus  Thank you for bringing this issue to our attention. Unfortunately, the change was necessary due to a security requirement mandated by Microsoft. While we recognize the impact this has had on your experience, ensuring the security of our systems and data is our top priority. Please rest assured that we have added this work item to our backlog. However, at this time, we do not have an estimated timeline for when a fix will be implemented. We appreciate your patience and understanding as we work towards a resolution.

**Comment by GruberMarkus** [comment]
@DivyaPatidar: Do you have any updates regarding this issue?

I would like to know if someone is working making addFileAttachmentAsync() available again in Outlook Web on premises, or when Exchange Server 2019 or Exchange Server Subscription Edition will support the API requirement set 1.8.

**Comment by sivachandran-msft** [comment]
Hi @GruberMarkus, as communicated earlier we have added this work item to our backlog. However, at this time, we do not have an estimated timeline for when a fix will be implemented. We appreciate your patience and understanding as we work towards a resolution.

**Comment by shabharinath** [comment]
Hi Team/ @sivachandran-msft  ,

The addFileAttachmentAsync() function in Office JS isn’t working in compose mode for Exchange Server 2019 On-Prem. This issue is critical as it affects our business and customers. The problem arises because the on-prem version only supports Office JS up to version 1.5, while addFileAttachmentAsync() requires version 1.8.

Could you please provide a fix for this?

(OR) let us know what would be the alternative for this addFileAttachmentAsync()?

Thanks

**Comment by mrharindra** [comment]
Hi @neprasad-microsoft  @sivachandran-msft 
Is there any update on this? 

Thanks
